pkg/scale/decode.go

```diff
@@ -523,10 +524,68 @@ func (ds *decodeState) decodeUint(dstv reflect.Value) (err error) {
 	return
 }
 
+var errorU32OutOfRange = errors.New("uint32 out of range")
+
+// decodeCompactUint32 will decode compact unsigned 32-bit integer
+func (ds *decodeState) decodeCompactUint32(dstv reflect.Value) (err error) {
```

Comment:
not sure why we need another function for this? Nevertheless `decodeUint` requires the overflow checking as well.  I'll sync up with you to clarify things.

Reply:
Removed `decodeCompactUint32`, and refactored `decodeUint` to add overflow checking.